user/client.go

```diff
@@ -0,0 +1,343 @@
+		Username       string   `json:"username,omitempty"`       // the user-assigned name for the login (usually an email address)
+		Emails         []string `json:"emails,omitempty"`         // the array of email addresses associated with this account
+		PasswordExists bool     `json:"passwordExists,omitempty"` // Does a password exist for the user?
+		Roles          []string `json:"roles,omitempty"`          // User roles
+		TermsAccepted  string   `json:"termsAccepted,omitempty"`  // When were the terms accepted
+		EmailVerified  bool     `json:"emailVerified,omitempty"`  // the user has verified the email used as part of signup
+	}
+
+	//TokenData is the data structure returned from a successful CheckToken query.
+	TokenData struct {
+		UserID   string // the UserID stored in the token
+		IsServer bool   // true or false depending on whether the token was a servertoken
+	}
+
+	//Permission for a user account
+	Permission map[string]interface{}
+	//Permissions for a user account
+	Permissions map[string]Permission
+	//UsersPermissions map of Permissions by userID
+	UsersPermissions map[string]Permissions
+)
+
+const (
+	xTidepoolServerName   = "x-tidepool-server-name"
+	xTidepoolServerSecret = "x-tidepool-server-secret"
+	xTidepoolSessionToken = "x-tidepool-session-token"
+	tidepoolClientSecret  = "TIDEPOOL_USER_CLIENT_SECRET"
+
+	userPath        = "/auth"
+	permissionsPath = "/access"
+	metaDataPath    = "/metadata"
+)
+
+//NewServicesClient returns and initailised ServicesClient instance
+func NewServicesClient() *ServicesClient {
+
+	var clientConfig *ClientConfig
+
+	config.FromJSON(&clientConfig, "userclient.json")
+
+	if clientConfig.Name == "" {
+		panic("ServicesClient requires a name to be set")
+	}
+	if clientConfig.Host == "" {
+		panic("ServicesClient requires a host to be set")
+	}
+
+	dur, err := time.ParseDuration(clientConfig.TokenRefreshInterval)
+	if err != nil {
+		log.Error("err getting the duration ", err.Error())
+	}
+	clientConfig.TokenRefreshDuration = dur
+
+	secret, err := config.FromEnv(tidepoolClientSecret)
+	if err != nil {
+		log.Error("err getting client secret ", err.Error())
+	}
+
+	return &ServicesClient{
+		httpClient: http.DefaultClient,
+		config:     clientConfig,
+		secret:     secret,
+		closed:     make(chan chan bool),
+	}
+}
+
+// Start starts the client and makes it ready for use.  This must be done before using any of the functionality
+// that requires a server token
+func (client *ServicesClient) Start() error {
+	if err := client.serverLogin(); err != nil {
+		log.Error("Problem with initial server token acquisition:", err.Error())
+	}
+
+	go func() {
+		for {
+			timer := time.After(time.Duration(client.config.TokenRefreshDuration))
+			select {
+			case twoWay := <-client.closed:
+				twoWay <- true
+				return
+			case <-timer:
+				if err := client.serverLogin(); err != nil {
+					log.Error("Error when refreshing server login:", err.Error())
```

Comment:
What happens here if this fails? It looks like it only will retry after another token refresh duration.